be/src/formats/orc/orc_chunk_writer.cpp

```diff
@@ -0,0 +1,736 @@
+            T value;
+            DecimalV3Cast::to_decimal_trivial<Type, T, false>(values[i], &value);
+            if constexpr (std::is_same_v<T, int128_t>) {
+                auto high64Bits = static_cast<int64_t>(value >> 64);
+                auto low64Bits = static_cast<uint64_t>(value);
+                decimal_orc_column.values[i] = orc::Int128{high64Bits, low64Bits};
+            } else {
+                decimal_orc_column.values[i] = value;
+            }
+        }
+        orc_column.hasNulls = true;
+    } else {
+        auto* values = ColumnHelper::cast_to_raw<DecimalType>(column)->get_data().data();
+        for (size_t i = 0; i < column_size; ++i) {
+            T value;
+            DecimalV3Cast::to_decimal_trivial<Type, T, false>(values[i], &value);
+            if constexpr (std::is_same_v<T, int128_t>) {
+                int64_t high64Bits = static_cast<int64_t>(value >> 64);
+                uint64_t low64Bits = static_cast<uint64_t>(value);
+                decimal_orc_column.values[i] = orc::Int128{high64Bits, low64Bits};
+            } else {
+                decimal_orc_column.values[i] = value;
+            }
+        }
+        memset(decimal_orc_column.notNull.data(), 1, column_size * sizeof(char));
+    }
+    decimal_orc_column.numElements = column_size;
+}
+
+void OrcChunkWriter::_write_decimals(orc::ColumnVectorBatch& orc_column, ColumnPtr& column, int precision, int scale) {
+    auto& decimal_orc_column = dynamic_cast<orc::Decimal128VectorBatch&>(orc_column);
+    auto column_size = column->size();
+
+    decimal_orc_column.resize(column_size);
+    decimal_orc_column.notNull.resize(column_size);
+
+    decimal_orc_column.precision = precision;
+    decimal_orc_column.scale = scale;
+
+    if (column->is_nullable()) {
+        auto c = ColumnHelper::as_raw_column<NullableColumn>(column);
+        auto* nulls = c->null_column()->get_data().data();
+        auto* values =
+                reinterpret_cast<int128_t*>(down_cast<DecimalColumn*>(c->data_column().get())->get_data().data());
+
+        for (size_t i = 0; i < column_size; ++i) {
+            if (nulls[i]) {
+                decimal_orc_column.notNull[i] = 0;
+                continue;
+            }
+            decimal_orc_column.notNull[i] = 1;
+            decimal_orc_column.values[i] = orc::Int128(values[i] >> 64, (values[i] << 64) >> 64);
+        }
+        orc_column.hasNulls = true;
+    } else {
+        auto* values = reinterpret_cast<int128_t*>(down_cast<DecimalColumn*>(column.get())->get_data().data());
+        for (size_t i = 0; i < column_size; ++i) {
+            decimal_orc_column.values[i] = orc::Int128(values[i] >> 64, (values[i] << 64) >> 64);
+        }
+        memset(decimal_orc_column.notNull.data(), 1, column_size * sizeof(char));
+    }
+    decimal_orc_column.numElements = column_size;
+}
+
+void OrcChunkWriter::_write_datetimes(orc::ColumnVectorBatch& orc_column, ColumnPtr& column) {
```

Comment:
You can use the name `_write_date` to ensure that function and type naming are consistent.